examples/tohoku_inversion/sources.py

```diff
@@ -0,0 +1,398 @@
+import firedrake as fd
+from thetis.configuration import *
+import thetis.utility as utility
+import ufl
+from okada import OkadaParameters, okada
+import abc
+import numpy
+
+
+__all__ = ["FiniteElementTsunamiSource", "RadialArrayTsunamiSource",
+           "BoxArrayTsunamiSource", "OkadaArrayTsunamiSource"]
+
+
+class TsunamiSource(FrozenConfigurable):
+    """
+    Base class for generating tsunami source conditions.
+    """
+
+    __metaclass__ = abc.ABCMeta
+
+    fault_length = PositiveFloat(560e3, help="Length of fault plane").tag(config=True)
+    fault_width = PositiveFloat(240e3, help="Width of fault plane").tag(config=True)
+    strike_angle = PositiveFloat(198, help="Angle of fault plane from North in degrees").tag(config=True)
+    fault_centroid_lon = BoundedFloat(143.6, bounds=(-180, 180), help="Longitude of the fault centroid").tag(config=True)
+    fault_centroid_lat = BoundedFloat(37.9, bounds=(-90, 90), help="Latitude of the fault centroid").tag(config=True)
```

Comment:
CI seemed to be stuck, so I fixed a minor TODO (previously here), rebased and pushed to get it going again.